pints/toy/_simple_egg_box.py

```diff
@@ -135,3 +156,17 @@ def sample(self, n):
         np.random.shuffle(x)
         return x
 
+    def distance(self, samples):
+        """
+        Calculates approximate mode-wise KL divergence (see `kl_divergence`)
+        """
+        return self.kl_divergence(samples)
+
+    def suggested_bounds(self):
```

Comment:
Probably best to add all these to a ToyProblem / ToyDistribution abstract class? Just so implementers know they're expected

Reply:
@MichaelClerx  Great idea! I have added an abstract ToyLogPDF base class.